sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala

```
@@ -561,4 +565,314 @@ case class TryAesDecrypt(
+    ZstdUtils.compress(
+      input1.asInstanceOf[Array[Byte]],
+      input2.asInstanceOf[Number].intValue(),
+      input3.asInstanceOf[Boolean]
+    )
+  }
+
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    val clz = ZstdUtils.getClass.getName.stripSuffix("$")
+    defineCodeGen(
+      ctx,
+      ev,
+      f = (input, level, streamingMode) => {
+        s"$clz.compress($input, $level, $streamingMode)"
+      }
+    )
+  }
+}
+
+// scalastyle:off line.size.limit
+/**
+ * A function that decompress input using Zstandard. On decompression failure, it throws an exception.
+ * If either argument is NULL, the return value is NULL.
+ */
+@ExpressionDescription(
+  usage = """
+    _FUNC_(expr) - Returns the decompressed value of `expr` using Zstandard.
+      Supports data compressed in both single-pass mode and streaming mode.
+      On decompression failure, it throws an exception.
+  """,
+  arguments = """
+    Arguments:
+      * expr - The binary value to decompress.
+  """,
+  examples = """
+    Examples:
+      > SELECT string(_FUNC_(unbase64("KLUv/SCCpQAAaEFwYWNoZSBTcGFyayABABLS+QU=")));
+       Apache Spark Apache Spark Apache Spark Apache Spark Apache Spark Apache Spark Apache Spark Apache Spark Apache Spark Apache Spark
+  """,
+  since = "4.0.0",
+  group = "misc_funcs")
+// scalastyle:on line.size.limit
+case class ZstdDecompress(input: Expression)
+    extends UnaryExpression with ImplicitCastInputTypes {
+  override def nullIntolerant: Boolean = true
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(BinaryType)
+
+  override def prettyName: String = "zstd_decompress"
+
+  override def child: Expression = input
+
+  override protected def withNewChildInternal(newChild: Expression): Expression = copy(newChild)
+
+  override def dataType: DataType = BinaryType
+
+  override protected def nullSafeEval(input: Any): Any = {
+    val result = ZstdUtils.decompress(input.asInstanceOf[Array[Byte]])
```

I think `ZstdUtils.decompress` should fail for invalid input.